openretina/insilico/tuning_analyses/response_gradient.py

```diff
@@ -0,0 +1,110 @@
+from torch import optim
+from torch import nn
+from copy import deepcopy
+import torch
+import numpy as np
+from openretina.insilico.stimulus_optimization.objective import SingleNeuronObjective
+
+class MeiAcrossContrasts(nn.Module):
+    """
+    PyTorch Module wrapper around a stimulus to allow tracking response gradients onto the stimulus.
+    """
+    def __init__(self, contrast_values: torch.TensorType, stim: torch.TensorType):
+        """
+        Initialize the ResponseGradient class.
+
+        Parameters:
+        contrast_values (torch.TensorType): A torch Tensor with 2 scalar values that the MEI green and UV channels will be multiplied with.
+        stim (torch.TensorType): MEI stimulus as torch Tensor.
+        """
+        super().__init__()
+        self.contrast_values = nn.Parameter(contrast_values, requires_grad=True)
+        self.stim = nn.Parameter(stim, requires_grad=False)
+
+    def forward(self):
+        return torch.mul(
+            torch.stack(
+                [torch.ones_like(self.stim[:, 0, ...]) * self.contrast_values[0],
+                 torch.ones_like(self.stim[:, 0, ...]) * self.contrast_values[1]
```

Comment:
```suggestion
                 torch.ones_like(self.stim[:, 1, ...]) * self.contrast_values[1]
```
maybe there's a typo above

Reply:
nvm, it does not matter and is not a bug.

Reply:
Thinking about this, I think you can make this more general such that it works for len(contrast_values) > 2, too:
```
import torch
stim = torch.rand((2,3,5,7))
big_contrast_values = contrast_values.view([len(contrast_values)] + [1 for _ in range(len(stim.shape)-2)])
scaled_stim = big_contrast_values * stim
```
E.g. the code above works. It only makes sense if you want to support models with more or less channels in the response gradient analysis, though.

Reply:
hm, I think it would be best to generalize this function in the next step to allow an arbitrary number of dimensions to be varied. I would make this a feature to add, and leave this basic function as is (i.e. tailored for 2 channels) for now